/*
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2.1 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this library; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *
 * ConfigMessages.proto
 * Defines the protocol buffers used to configure the Artnet device
 * Copyright (C) 2005 - 2008 Simon Newton
 */

package ola.plugin.artnet;

message OptionsRequest {
  optional string short_name = 1;
  optional string long_name = 2;
  optional int32 subnet = 3;
  optional int32 net = 4;
}

message OptionsReply {
  required uint32 status = 1;
  required string short_name = 2;
  required string long_name = 3;
  required int32 subnet = 4;
  required int32 net = 5;
}

message Request {
  enum RequestType {
    ARTNET_OPTIONS_REQUEST = 1;
  }

  required RequestType type = 1;
  optional OptionsRequest options = 2;
}

message Reply {
  enum ReplyType {
    ARTNET_OPTIONS_REPLY = 1;
  }
  required ReplyType type = 1;

  optional OptionsReply options = 2;
}
